### DISCUSSION TITLE
ESPHome and Wifi/RS485

### DISCUSSION THREAD

**Discussion by whizkidTRW** [aqualinkd/AqualinkD#203] [General]
I have just installed home assistant OS on a PC and have ordered some ESPHome devices. One is for my Century VS pump but the other I’d like to use with the Hayward SwimPure chlorine generator. Is there an option to connect to the RS485 via ESPHome/Wi-Fi instead of USB?

**Comment by ballle98** [comment]
AqualinkD is a complicated proprietary serial protocol over RS485 to web GUI & MQTT converter.  If you put the RS485 into another device you are left with a protocol that nothing besides aqualinkd can understand if you connect ESP or RP2040.

The simpelest thing would be to convert usb/tty to tcp/ip and run aqualinkd remotely - see https://unix.stackexchange.com/questions/100845/serial-data-over-ethernet-on-a-linux-box

You can run aqualinkd on a Pi Zero so not sure its worth messing with a microcontroller for this
